app/javascript/src/Applications/ApplicationForm.jsx

```diff
@@ -0,0 +1,161 @@
+// @flow
+
+import React, { useState, useEffect } from 'react'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+import './applicationForm.scss'
+
+type ApplicationPlan = {
+  id: number,
+  name: string,
+}
+
+type ServicePlan = {
+  id: number,
+  name: string,
+  default: boolean
+}
+
+type Props = {
+  plans: ApplicationPlan[],
+  servicesContracted: number[],
+  relationServiceAndServicePlans: {[number]: ServicePlan[]},
+  relationPlansServices: {[number]: number},
+  servicePlanContractedForService: {[number]: ServicePlan},
+  servicePlansAllowed: boolean
+}
+
+const ApplicationForm = ({
+  plans, servicesContracted, relationServiceAndServicePlans,
+  relationPlansServices, servicePlanContractedForService, servicePlansAllowed
+}: Props) => {
+  const [selectedPlan, setSelectedPlan] = useState(plans[0])
+  const [term, setTerm] = useState(selectedPlan.name)
+
+  function checkSelectedPlan () {
+    const serviceId = getServiceIdOfPlanId(selectedPlan.id)
+    const servicePlans = getServicePlansForService(serviceId)
+
+    enableForm()
+
+    if (servicesContracted.indexOf(serviceId) > -1) {
+      const { id, name } = getContractedServicePlanForService(serviceId)
+      $('#cinstance_service_plan_id').html(`<option value="${id}"> ${name} </option>`)
+      disableField('#cinstance_service_plan_id')
+    } else if (servicePlans.length !== 0) {
+      setServicePlansSelectOptions(servicePlans)
+    } else {
+      $('#cinstance_service_plan_id').html('<option> No service plan for the application plan </option>')
+      disableForm()
+    }
+  }
+
+  function getServiceIdOfPlanId (id: number): number {
+    return relationPlansServices[id]
+  }
+
+  function getServicePlansForService (serviceId: number): ServicePlan[] {
+    return relationServiceAndServicePlans[serviceId]
+  }
+
+  function getContractedServicePlanForService (serviceId: number): ServicePlan {
+    return servicePlanContractedForService[serviceId]
+  }
+
+  function enableForm () {
+    $('#link-help-new-application-service').toggle(false)
+    enableField('#submit-new-app')
+    enableField('#cinstance_service_plan_id')
+  }
+
+  function disableForm () {
+    $('#link-help-new-application-service').toggle(true)
+    disableField('#submit-new-app')
+    disableField('#cinstance_service_plan_id')
+  }
+
+  function enableField (field: string) {
+    $(field).removeAttr('disabled')
```

Comment:
Maybe having a sole `toggleField` or `toggleDisabled` and using `Element.toggleAttribute`.

Reply:
In progress